tests/mcs/tso/server_test.go

```diff
@@ -222,51 +227,55 @@ func (suite *APIServerForwardTestSuite) TearDownTest() {
 }
 
 func (suite *APIServerForwardTestSuite) TestForwardTSORelated() {
-	leader := suite.cluster.GetServer(suite.cluster.WaitLeader())
-	suite.NoError(leader.BootstrapCluster())
-	suite.addRegions()
 	// Unable to use the tso-related interface without tso server
 	suite.checkUnavailableTSO()
 	// can use the tso-related interface with tso server
-	_, cleanup := mcs.StartSingleTSOTestServer(suite.ctx, suite.Require(), suite.backendEndpoints)
+	s, cleanup := mcs.StartSingleTSOTestServer(suite.ctx, suite.Require(), suite.backendEndpoints)
+	serverMap := make(map[string]bs.Server)
+	serverMap[s.GetAddr()] = s
+	mcs.WaitForPrimaryServing(suite.Require(), serverMap)
 	suite.checkAvailableTSO()
 	cleanup()
 }
 
 func (suite *APIServerForwardTestSuite) TestForwardTSOWhenPrimaryChanged() {
-	leader := suite.cluster.GetServer(suite.cluster.WaitLeader())
-	suite.NoError(leader.BootstrapCluster())
-	suite.addRegions()
-	serverMap := make(map[string]*tsosvr.Server)
+	serverMap := make(map[string]bs.Server)
 	for i := 0; i < 3; i++ {
 		s, cleanup := mcs.StartSingleTSOTestServer(suite.ctx, suite.Require(), suite.backendEndpoints)
 		defer cleanup()
 		serverMap[s.GetAddr()] = s
 	}
+	mcs.WaitForPrimaryServing(suite.Require(), serverMap)
 
 	// can use the tso-related interface with new primary
 	oldPrimary, exist := suite.pdLeader.GetServer().GetServicePrimaryAddr(suite.ctx, utils.TSOServiceName)
 	suite.True(exist)
 	serverMap[oldPrimary].Close()
+	delete(serverMap, oldPrimary)
 	time.Sleep(time.Duration(utils.DefaultLeaderLease) * time.Second) // wait for leader lease timeout
+	mcs.WaitForPrimaryServing(suite.Require(), serverMap)
 	primary, exist := suite.pdLeader.GetServer().GetServicePrimaryAddr(suite.ctx, utils.TSOServiceName)
 	suite.True(exist)
 	suite.NotEqual(oldPrimary, primary)
 	suite.checkAvailableTSO()
 
 	// can use the tso-related interface with old primary again
-	_, cleanup := mcs.StartSingleTSOTestServer(suite.ctx, suite.Require(), suite.backendEndpoints, oldPrimary)
+	newPrimary := tempurl.Alloc()
```

Comment:
According to the comment, we should test to start the previous server with the same address to see if it works, instead of using a new address

Reply:
use `tempurl.Alloc()` change test behavior

Reply:
changed back